File: docs/examples/go/src/access_control_create_token.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
	model "github.com/reductstore/reduct-go/model"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Create a token with read/write access, a 1 hour inactivity TTL,
	// and an IP allowlist for local access.
	permissions := model.TokenPermissions{
		FullAccess: false,
		Read:       []string{"example-bucket"},
		Write:      []string{"example-bucket"},
	}

	createResp, err := client.CreateTokenWithOptions(context.Background(), "new-token", model.TokenCreateOptions{
		Permissions: permissions,
		TTL:         3600,
		IPAllowlist: []string{"127.0.0.1", "::1"},
	})
	if err != nil {
		panic(err)
	}

	// Print the generated token
	println("generated token:", createResp.Value)
}

File: versioned_docs/version-1.19.x/examples/py/src/access_control_create_token.py
from reduct import Client, Permissions


async def create_token():
    # Create a client with the base URL and API token
    async with Client("http://localhost:8383", api_token="my-token") as client:
        # Create a token with read/write access, a 1 hour inactivity TTL,
        # and an IP allowlist for local access.
        permissions = Permissions(
            full_access=False,
            read=["example-bucket"],
            write=["example-bucket"],
        )
        token = await client.create_token(
            "new-token",
            permissions,
            ttl=3600,
            ip_allowlist=["127.0.0.1", "::1"],
        )
        print(f"generated token: {token}")


if __name__ == "__main__":
    import asyncio

    asyncio.run(create_token())
